src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx

```diff
@@ -444,20 +442,53 @@ com.example.MyAmplifyApp I/MyAmplifyApp: Added Todo with id: SOME_TODO_ID
 
 <InlineFilter filters={["swift"]}>
 
-Go to your **ContentView.swift** and add a button to create a new todo:
+Create a new file called `TodoViewModel.swift` and the `createTodo` function with the following code:
 
-```swift title="ContentView.swift"
-import SwiftUI
-// highlight-next-line
+```swift title="TodoViewModel.swift"
 import Amplify
 
+@MainActor
+class TodoViewModel: ObservableObject {
```

Comment:
You need to import `Foundation` for `ObservableObject` to be accessible
<img width="656" alt="image" src="https://github.com/aws-amplify/docs/assets/97059974/766d37f4-e8c1-438f-8307-f7461608bce1">

Reply:
thanks!